engine/src/flutter/lib/ui/semantics/semantics_update_builder_unittests.cc

```diff
@@ -131,5 +131,48 @@ TEST_F(SemanticsUpdateBuilderTest, CanHandleSemanticsRole) {
   DestroyShell(std::move(shell), task_runners);
 }
 
+TEST_F(SemanticsUpdateBuilderTest, CanHandleSemanticsLocale) {
+  auto message_latch = std::make_shared<fml::AutoResetWaitableEvent>();
+
+  auto nativeSemanticsUpdate = [message_latch](Dart_NativeArguments args) {
+    auto handle = Dart_GetNativeArgument(args, 0);
+    intptr_t peer = 0;
+    Dart_Handle result = Dart_GetNativeInstanceField(
+        handle, tonic::DartWrappable::kPeerIndex, &peer);
+    ASSERT_FALSE(Dart_IsError(result));
+    SemanticsUpdate* update = reinterpret_cast<SemanticsUpdate*>(peer);
+    SemanticsNodeUpdates nodes = update->takeNodes();
+    ASSERT_EQ(nodes.size(), (size_t)1);
+    auto node = nodes.find(0)->second;
+    // Should match the updateNode in ui_test.dart.
+    ASSERT_EQ(node.locale, "es-MX");
+    message_latch->Signal();
+  };
+
+  Settings settings = CreateSettingsForFixture();
+  TaskRunners task_runners("test",                  // label
+                           GetCurrentTaskRunner(),  // platform
+                           CreateNewThread(),       // raster
+                           CreateNewThread(),       // ui
+                           CreateNewThread()        // io
+  );
+
+  AddNativeCallback("SemanticsUpdate",
+                    CREATE_NATIVE_ENTRY(nativeSemanticsUpdate));
+
+  std::unique_ptr<Shell> shell = CreateShell(settings, task_runners);
+
+  ASSERT_TRUE(shell->IsSetup());
+  auto configuration = RunConfiguration::InferFromSettings(settings);
+  configuration.SetEntrypoint("sendSemanticsUpdateWithLocale");
+
+  shell->RunEngine(std::move(configuration), [](auto result) {
```

Comment:
FYI this PR has a lot of usage of `auto` which are against the style guide:

https://google.github.io/styleguide/cppguide.html#Type_deduction